clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -169,6 +210,14 @@ public synchronized boolean updateRequested() {
         return this.needUpdate;
     }
 
+    /**
+     * Return the cached partition info if it exists and a newer leader epoch isn't known about.
+     */
+    public synchronized Optional<PartitionInfo> getLeaderInfo(TopicPartition topicPartition) {
+        return Optional.ofNullable(lastSeenLeaderEpochs.get(topicPartition))
```

Comment:
Hmm.. If `lastSeenLeaderEpochs` doesn't have the partition, should we return the cached `PartitionInfo`? Otherwise, it wouldn't work on an older cluster which does not expose leader epoch in the Metadata API.

Reply:
Fixed